File: C++/2485. Find the Pivot Integer.cpp
class Solution {
public:
    int pivotInteger(int n) {
        int low = 1, high = n;
        int totalSum = (n * (n + 1)) / 2;
        while (low <= high) {
            int mid = low + (high - low) / 2;
            int currSum = (mid * (mid + 1)) / 2;
            if (totalSum - currSum + mid == currSum)
                return mid;
            else if (currSum < totalSum / 2)
                low = mid + 1;
            else
                high = mid - 1;
        }
        return -1;
    }
};

File: JAVA/2485. Find the Pivot Integer.java
class Solution {
    public int pivotInteger(int n) {
        int sum=0;
        for(int i=1;i<=n;i++){
            sum+=i;
        }
        int p=0;
        for(int i=1;i<=n;i++){
p+=i;
if(p==sum){
    return i;
}
sum=sum-i;
        }
        return -1;
    }
}
